src/providers/citation.ts

```diff
@@ -35,10 +35,32 @@ export class Citation {
             this.citationInBib[bibPath].forEach(item => items.push(item))
         })
 
+        const configuration = vscode.workspace.getConfiguration('latex-workshop')
         this.suggestions = items.map(item => {
             const citation = new vscode.CompletionItem(item.key, vscode.CompletionItemKind.Reference)
             citation.detail = item.title
-            citation.filterText = `${item.author} ${item.title} ${item.journal}`
+            switch (configuration.get('citation_intellisense_label') as string) {
+                case 'bibtex key':
+                default:
+                    citation.label = item.key
+                    break
+                case 'title':
+                    if (item.title) {
+                        citation.label = item.title as string
+                    } else {
+                        citation.label = item.key
+                    }
+                    break
+                case 'authors':
+                    if (item.author) {
+                        citation.label = item.author as string
```

Comment:
Perhaps here too. Basic idea being: 

1. by default we show you the bibtex key for sure and try and show as much as the title as possible in `detail`.
2. You can override this if you rather the view be maximised for either `title` or `author`, but in these cases we should hide the `detail` to give you as much room as possible for the thing you have selected.